descope/internal/mgmt/user.go

```diff
@@ -420,15 +420,15 @@ func (u *user) RemoveTenantRoles(ctx context.Context, loginID string, tenantID s
 	return unmarshalUserResponse(res)
 }
 
-func (u *user) SetPassword(ctx context.Context, loginID string, password string) error {
+func (u *user) SetPassword(ctx context.Context, loginID string, password string, persistPassword bool) error {
```

Comment:
I think I would add a new function here, so it will not break compilation
But not sure, maybe hear from another person as well

Reply:
I dont think we need a new function here as it will be confusing (as its simple logic which have two function - its too much overloaded),
I checked in DD and there is no usage of this API by the go-sdk so I think we should be ok here.

Reply:
I think one method is okay. I've seen customers getting confused by the default behavior being that hte password is set but is "expired".

Maybe change the signature to `SetPassword(ctx context.Context, loginID string, password string, temporary bool)`? Then the default behavior with `false` value for the `temporary` flag is what you would expect a function called `SetPassword` to do.

Another alternative, two functions like Aviad said, `SetTemporaryPassword` and `SetActivePassword`? @itaihanski wdyt?

Reply:
I like @shilgapira 's idea. "Persist" is a bit ambiguous. I would go with the temporary / active naming.

Reply:
noting that `temporary bool` is not ideal 
because the default behavior should be temporary (if user didn't specify anything)

though in go, user must provide a value, I think that 
a) providing `false` is more natural/passive imo than providing `true`
b) reg other SDKs implementations -  though consistency with other descope SDK (node, python) is not a must, inconsistency creates complexity

Reply:
A boolean flag is not very readable or understandable for customers IMO. Customers have already gotten confused over this. Being as explicit as possible is preferable. SetTemporaryPassword / SetActivePassword should be the way to go.

Reply:
I dont think its a good idea to create two function for such basic logic, lets keep the SetPassword(..) API (which already exist for all the sdks) API + a boolean flag called "setActive" so it will be clear whats going on with the password.

Reply:
so discussed this once more with @shilgapira and after some thoughts we came to a conclusion that it will be best to go with the proposal to have two new functions (will mark the old one as deprecated):
SetTemporaryPassword(..)
SetActivePassword(..)